packages/Skeleton/withSkeleton.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { SkeletonContext } from './SkeletonProvider'
+
+class SkeletonRenderer extends React.Component {
+  static contextType = SkeletonContext
+  static propTypes = {
+    skeleton: PropTypes.func.isRequired,
+    render: PropTypes.func.isRequired,
+  }
+
+  render() {
+    return (
+      <SkeletonContext.Consumer>
+        {show => (show ? this.props.skeleton() : this.props.render())}
+      </SkeletonContext.Consumer>
+    )
+  }
+}
+
+const getName = comp => comp.displayName || comp.name || 'Component'
+
+const withSkeleton = SkeletonComponent => {
+  return Component => {
+    return function wrapper({ skeleton, ...props }) {
```

Comment:
better to have more descriptive function name in case of error it would be easier to know where it came from. I think it would be nicer to call this `withSkeletonWrapper` instead of `wrapper`

Reply:
I already manually set the display name to to `withSkeleton(<ComponenetName>)` so that it will appear like this in the tree: 

![image](https://user-images.githubusercontent.com/32303524/50646375-e1fa1800-0f43-11e9-9086-25875da22bc9.png)